### DISCUSSION TITLE
What is the underlying need for aliasing @prisma/client with Cloudflare?

### DISCUSSION THREAD

**Discussion by RyKilleen** [prisma/prisma#11823] [Q&A] [answered]
Per the [Prisma documents here](https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-cloudflare-workers#4-set-up-webpack), when using Prisma with Cloudflare you need to use the alias plugin.

I'm trying to understand a little more deeply _why_ this is needed. Right now the Remix community is running into a challenge that's two-fold: Remix uses esbuild (which has esbuild-alias-plugin) but you can't currently extend the underlying esbuild process, meaning you have to overwrite it.

I'm hoping to understand the requirement here a little more deeply so as to hopefully find some other way around this challenge.  If anyone can shed light I'd appreciate it, thanks!

**Comment by millsp** [comment, marked as answer]
Hey @RyKilleen,

This config was needed because when you bundle for the browser (eg. workers, vercel edge), then those bundlers involved (usually webpack) will prioritize bundling via browser field entry points (found in package.json). The problem here is that we want exactly that, but the browser field for the prisma client has already been borrowed.

Back in the days, we released a new a "browser" build so that folks could import generated artifacts and types without the client runtime itself (eg. enums to be re-used in react apps...). Since we cannot break that, we needed a workaround.

I'm fairly sure you can do that with esbuild or others. With esbuild you would just rewrite the import to point directly to your generated client `index.js`, and thus bypassing the browser fields resolution of the bundler. That is what this webpack config means.

That's something that will get fixed I guess (but not planned yet). A neat solution would be to have a `@prisma/client/edge` import.

Let me know if you'd need help to set that up :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
